### ISSUE TITLE
java.lang.IllegalArgumentException: Cannot copy between a TensorFlowLite tensor with shape [1, 11] and a Java object with shape [1, 11, 4].

### ISSUE THREAD

**Issue by aakashganga** [shaqian/flutter_tflite#41]
I trained a Mobilenet model using transfer learning with Keras. I then converted the keras h5 model to tflite using toco and replaced the mobilnet model and the labels with my labels. I have 11 classes in my model. But, while running, I am getting the above error. Here is the full trace.
``
E/AndroidRuntime(21599): FATAL EXCEPTION: AsyncTask #2
E/AndroidRuntime(21599): Process: com.example.examplerealtime, PID: 21599
E/AndroidRuntime(21599): java.lang.RuntimeException: An error occurred while executing doInBackground()
E/AndroidRuntime(21599):        at android.os.AsyncTask$3.done(AsyncTask.java:354)
E/AndroidRuntime(21599):        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
E/AndroidRuntime(21599):        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
E/AndroidRuntime(21599):        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
E/AndroidRuntime(21599):        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
E/AndroidRuntime(21599):        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
E/AndroidRuntime(21599):        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
E/AndroidRuntime(21599):        at java.lang.Thread.run(Thread.java:764)
E/AndroidRuntime(21599): Caused by: java.lang.IllegalArgumentException: Cannot copy between a TensorFlowLite tensor with shape [1, 11] and a Java object with shape [1, 11, 4].
E/AndroidRuntime(21599):        at org.tensorflow.lite.Tensor.throwIfShapeIsIncompatible(Tensor.java:282)
E/AndroidRuntime(21599):        at org.tensorflow.lite.Tensor.throwIfDataIsIncompatible(Tensor.java:249)
E/AndroidRuntime(21599):        at org.tensorflow.lite.Tensor.copyTo(Tensor.java:141)
E/AndroidRuntime(21599):        at org.tensorflow.lite.NativeInterpreterWrapper.run(NativeInterpreterWrapper.java:161)
E/AndroidRuntime(21599):        at org.tensorflow.lite.Interpreter.runForMultipleInputsOutputs(Interpreter.java:275)
E/AndroidRuntime(21599):        at sq.flutter.tflite.TflitePlugin$RunSSDMobileNet.runTflite(TflitePlugin.java:622)
E/AndroidRuntime(21599):        at sq.flutter.tflite.TflitePlugin$TfliteTask.doInBackground(TflitePlugin.java:419)
E/AndroidRuntime(21599):        at sq.flutter.tflite.TflitePlugin$TfliteTask.doInBackground(TflitePlugin.java:393)
E/AndroidRuntime(21599):        at android.os.AsyncTask$2.call(AsyncTask.java:333)
E/AndroidRuntime(21599):        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
``

**Comment by shaqian** [comment]
Is your model trained with SSD? SSD should have 4 outputs but your model only have one. Rather than `detectObjectOnImage` can you please try `runModelOnImage`?

Thanks,
Qian

**Comment by aakashganga** [comment]
Hi Qian, the model is trained with Mobilenet. Here is my code
`#base_model=MobileNet(weights='imagenet',include_top=False) #imports the mobilenet model and discards the last 1000 neuron layer.
base_model = MobileNet(include_top=False, weights="imagenet", input_tensor=Input(shape=(224,224,3)), input_shape=(224,224,3))
x=base_model.output
x=GlobalAveragePooling2D()(x)
x=Dense(1024,activation='relu')(x) #we add dense layers so that the model can learn more complex functions and classify for better results.
x=Dense(1024,activation='relu')(x) #dense layer 2
x=Dense(512,activation='relu')(x) #dense layer 3
preds=Dense(11,activation='softmax')(x) #final layer with softmax activation
model=Model(inputs=base_model.input,outputs=preds)
`  I think you are saying that mobilenet is different from mobilenet ssd. Is that right?

**Comment by aakashganga** [comment]
Hi Qian, per you recommendation, I added the following code to my camera.dart
` } else if (widget.model == "SSDMobileNet") {
              Tflite.runModelOnFrame(
                bytesList: img.planes.map((plane) {
                  return plane.bytes;
                }).toList(),
                imageHeight: img.height,
                imageWidth: img.width,
                numResults: 2,
              ).then((recognitions) {
                int endTime = new DateTime.now().millisecondsSinceEpoch;
                print("Detection took ${endTime - startTime}");

                widget.setRecognitions(recognitions, img.height, img.width);

                isDetecting = false;
              });`

But, now, I am getting
 java.lang.IllegalArgumentException: Cannot copy between a TensorFlowLite tensor with shape [1, 11] and a Java object with shape [1, 7, 7, 80].

**Comment by aakashganga** [comment]
For both Mobilenet and yolo, I get the following error when I replace the existing tflite and labels files with my files.

> java.lang.IllegalArgumentException: Cannot convert between a TensorFlowLite tensor with type UINT8 and a Java object of type [[F (which is compatible with the TensorFlowLite type FLOAT32).

**Comment by aakashganga** [comment]
Hi @shaqian any thoughts into this? Basically, I am trying to understand how I can use my transfer learning Keras model with this framework.

**Comment by shaqian** [comment]
Hi @aakashganga,

Can you use `tfLite.getOutputTensor(0).shape()` to get the shape of output tensor of your model? For image classification, the output size should be the number of classes, so if you trained 11 classes, the shape should be [1, 11] not [1, 7, 7, 80].

The `IllegalArgumentException` would happen when date type doesn't match. I don't know if it's happening with input tensor or output tensor. Can you share the call stack?

Thanks,
Qian

**Comment by shaqian** [comment]
I'm archiving this thread. Feel free to reopen if you have further questions.

Thanks,
Qian
